### ISSUE TITLE
How to setup when deployed

### ISSUE THREAD

**Issue by chris-herring** [microsoft/playwright-dotnet#1692]
I have an ASP Core project working fine locally with Playwright. I've published it and deployed to a server however am unable to get it running there. 
The logs are outputting
`    Executable doesn't exist at C:\Windows\system32\config\systemprofile\AppData\Local\ms-playwright\chromium-901522\chrome-win\chrome.exe`

I've run 
`dotnet tool install --global Microsoft.Playwright.CLI
`
Is there something else I've missed?

**Comment by chris-herring** [comment]
> After installing the CLI, you need to execute `playwright install` to install the browsers.

I get the following error when I try that. There is no project file as it's a published build of the app.
`[91mCouldn't find project using Playwright. Ensure a project or a solution exists in C:\Octopus\Applications\..., or provide another path using -p.[0m`

**Comment by chris-herring** [comment]
Looking at the code it requires a csproj or sln file. So for a published build is the best solution to bundle the browser(s) in the deployment package? If so what's the recommended way to do this and tell Playwright to use the bundle browsers?

**Comment by avodovnik** [comment]
There is no way to bundle the browsers at the moment. 

The next best option, at the moment, is to run `playwright.ps` script that is included in the build output of the project. You should be able to invoke the install through that. 

The last option, you can invoke the node tooling directly, i.e. `bin\Debug\net5.0\.playwright\node\win32_x64\playwright.cmd` where obviously you replace the platform with the relevant one you're deploying to.

We're working on making this experience better though.

**Comment by avodovnik** [comment]
Oh, one more thing, if you *have* dotnet installed (i.e., it's not a self-contained app), you can also do  `dotnet .\Microsoft.Playwright.dll install` as per my comment here: https://github.com/microsoft/playwright-dotnet/issues/1665#issuecomment-896555590

**Comment by report-scraper** [comment]
I am having the same type of issues trying to get the library to work when it is published as part of a windows service to a windows server.  The SDK is not installed on the windows server so the global tools are not available.

When I try to install the playwright by using the powershell script I am seeing the following error:

``
.\playwright.ps1

Unable to find type [Microsoft.Playwright.Program].
At xxx\playwright.ps1:2 char:1
+ [Microsoft.Playwright.Program]::Main($args)
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (Microsoft.Playwright.Program:TypeName) [], RuntimeException
    + FullyQualifiedErrorId : TypeNotFound

Is there something else I need to do to get the generated script running correctly?

**Comment by chris-herring** [comment]
I got it working but it wasn't simple. Here's what I did incase anyone else hits this.

First step was trying to install the browsers.  

The CLI didnt work due to no project or solution file. I uninstalled the CLI at some point but I don't remember why now.

The Playwright.ps file did not exist in my published files however I copied the file over manually however it resulted in the following error:

```
Unable to find type [Microsoft.Playwright.Program].
At C:\Octopus\Applications\CivilSupport\Production\Contrack.App\1.0.0.73\playwright.ps1:2 char:1
+ [Microsoft.Playwright.Program]::Main($args)
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : InvalidOperation: (Microsoft.Playwright.Program:TypeName) [], RuntimeException
+ FullyQualifiedErrorId : TypeNotFound
```


I tried `dotnet .\Microsoft.Playwright.dll install` however it failed to run as a self-contained app because Microsoft.Playwright.runtimeconfig.json was not found.

I did manage to install using node - I think both `npx playwright install` and `playwright.cmd install` worked for me.

However it still did not work because the browsers were installed to the user profile and the app was running under an IIS app pool identity and was throwing this error:
`Executable doesn't exist at C:\Windows\system32\config\systemprofile\AppData\Local\ms-playwright\chromium-901522\chrome-win\chrome.exe`

So the next step was getting the app to find the executable. I set the `PLAYWRIGHT_BROWSERS_PATH `environment variable to "C\Playwright" and installed the browsers again and they installed to the correct location. However whatever I did I could not get the app to pick up that environment variable. I set the environment variable in the system environment variables to ensure that app pool identity account would pick it up but it kept referencing the old path. 

I ended up adding an app setting to my app for the executable path. In the app I then pass this path in the BrowserTypeLaunchOptions when launching the browser in the app and it all works.
